### ISSUE TITLE
(Assets): Allow custom Docker image build arguments

### ISSUE THREAD

**Issue by IronCrystal** [aws/aws-cdk#29111]
### Describe the feature

I can see there have been many feature requests in the past for support for different arguments to be added to the docker build command.  Isn't it time to just allow the developer to specify their own custom arguments?

Specifically I need access to the `--build-context` argument, but I think it would be better suited to support any custom arguments.

### Use Case

I have a situation where I need to use the `--build-context` arguments when building.

I have a monorepo with shared packages directory in the root, along with Dockerfiles in each microservice directory.

I need to be able to support a context that includes the files in the microservice sub directory, along with the shared packages directory.

I am able to do this with docker build by hand using the `--build-context` argument since docker supports multiple contexts now.  Docker compose also supports this with the `additional_contexts` property.

I don't see any way in the current CDK docker image implementation to support this.

### Proposed Solution

A new option that can be passed into the ecrAssets.DockerImageAsset class.

Maybe `customArgs` which supports a string array, or just a string.

It would be added to the `packages/cdk-assets/lib/private/docker.ts` build function.

```
public async build(options: BuildOptions) {
    const buildCommand = [
      'build',
      ...flatten(Object.entries(options.buildArgs || {}).map(([k, v]) => ['--build-arg', `${k}=${v}`])),
      ...flatten(Object.entries(options.buildSecrets || {}).map(([k, v]) => ['--secret', `id=${k},${v}`])),
      ...options.buildSsh ? ['--ssh', options.buildSsh] : [],
      '--tag', options.tag,
      ...options.target ? ['--target', options.target] : [],
      ...options.file ? ['--file', options.file] : [],
      ...options.networkMode ? ['--network', options.networkMode] : [],
      ...options.platform ? ['--platform', options.platform] : [],
      ...options.outputs ? options.outputs.map(output => [`--output=${output}`]) : [],
      ...options.cacheFrom ? [...options.cacheFrom.map(cacheFrom => ['--cache-from', this.cacheOptionToFlag(cacheFrom)]).flat()] : [],
      ...options.cacheTo ? ['--cache-to', this.cacheOptionToFlag(options.cacheTo)] : [],
      ...options.cacheDisabled ? ['--no-cache'] : [],
      '.',
    ];
    await this.execute(buildCommand, {
      cwd: options.directory,
      quiet: options.quiet,
    });
  }
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.118.0

### Environment details (OS name and version, etc.)

mac os/ubuntu

**Comment by pahud** [comment]
>Maybe customArgs which supports a string array, or just a string.

Yes this might be a good option. Please help us prioritize with 👍  and we welcome any pull requests.

**Comment by magiconair** [comment]
Got exactly the same problem with `--build-context` and shared package directory. Can't build this image with CDK the same way I build this on the command line.

**Comment by jakajancar** [comment]
There is no chance you will manage to mirror all the happenings in the docker/buildkit ecosystem so please give us an escape hatch so we can get our jobs done :)

**Comment by regoawt** [comment]
We are also suffering from the inability to pass shared libs using `--build-context`. Has anyone come up with a workaround for now?
